File: 蓝桥杯练习试题/4算法提高/输出九九乘法表/1.c
#include <stdio.h>
void print(int n)
{
	if(n==10) return ;
	int i,j;
	printf("%d ",n);
	for(i=1;i<=n;i++)
	printf("%d ",n*i);
	printf("\n"); 
	print(n+1);
}

int main()
{
printf("Nine-by-nine Multiplication Table\n");
printf("--------------------------------------\n");
printf("1 2 3 4 5 6 7 8 9\n");
printf("--------------------------------------\n");
print(1);
printf("--------------------------------------");
	return 0;
}

File: 蓝桥杯练习试题/4算法提高/输出九九乘法表/1.java
public class Main{

	public static void main(String args[]){
	
		System.out.println("Nine-by-nine Multiplication Table");
		System.out.println("--------------------------------------");	
		System.out.println("1 2 3 4 5 6 7 8 9");	
		System.out.println("--------------------------------------");	
			for(int i=1;i<=9;i++){
				for(int j=1;j<=i;j++){
					if(j<=1){
					System.out.print(i+" "+i*j+" ");
					}
					else{
						System.out.print(i*j+" ");
					}
				}
				System.out.println();
			}System.out.println("--------------------------------------");
}
}
